src/game/game_state.cpp

```diff
@@ -50,7 +60,13 @@ void GameState::update() {
     scene_manager.get_current_scene()->update();
 
     if (curr_phase == &dungeon_phase) {
```

Comment:
This is so gross, why are we not handling this in the actual `DungeonPhase::update()`?

If it's an issue of the `DungeonPhase` not knowing enough information, we should be passing it. We should think about creating a `Context` class that is owned by `GameState`, which contains phase-independent information which the active `Phase` can mutate.